compose/cli/log_printer.py

```diff
@@ -176,8 +177,13 @@ def build_log_generator(container, log_args):
 
 
 def wait_on_exit(container):
-    exit_code = container.wait()
-    return "%s exited with code %s\n" % (container.name, exit_code)
+    try:
+        exit_code = container.wait()
+        return "%s exited with code %s\n" % (container.name, exit_code)
+    except APIError as e:
+        return "Unexpected API error for %s (HTTP code %s)\n" % (
+            container.name, e.response.status_code
+        )
```

Comment:
Can we include the response body as well, which might contain a useful error message?